src/routes/transactions/helpers/kiln-dedicated-staking.helper.ts

```diff
@@ -0,0 +1,59 @@
+import { IStakingRepository } from '@/domain/staking/staking.repository.interface';
+import { StakingRepositoryModule } from '@/domain/staking/staking.repository.module';
+import {
+  TransactionFinder,
+  TransactionFinderModule,
+} from '@/routes/transactions/helpers/transaction-finder.helper';
+import { Inject, Injectable, Module } from '@nestjs/common';
+import { toFunctionSelector } from 'viem';
+
+@Injectable()
+export class KilnDedicatedStakingHelper {
+  constructor(
+    private readonly transactionFinder: TransactionFinder,
+    @Inject(IStakingRepository)
+    private readonly stakingRepository: IStakingRepository,
+  ) {}
+
+  public async findDeposit(args: {
+    chainId: string;
+    to?: `0x${string}`;
+    data: `0x${string}`;
+  }): Promise<{
+    to: `0x${string}`;
+    data: `0x${string}`;
+  } | null> {
+    const selector = toFunctionSelector('function deposit() external payable');
+    const transaction = this.transactionFinder.findTransaction(
+      (transaction) => transaction.data.startsWith(selector),
+      args,
+    );
+
+    if (!transaction?.to) {
+      return null;
+    }
+
+    const deployment = await this.stakingRepository
+      .getDeployment({
```

Comment:
We should add a caching later to the staking datasource as these calls are required multiple times per transaction.